src/care/host_device_ptr.h

```diff
@@ -208,6 +208,13 @@ namespace care {
 #endif
       { return MA::operator[](i); }
 
+#if defined(CARE_LEGACY_COMPATIBILITY_MODE)
+     template<typename Idx>
+     inline CARE_HOST_DEVICE T& operator[](const Idx i) const {
```

Comment:
This should return const T&

Reply:
I will probably kill this anyway after Peter said that was ok in the meeting.

But in any case, when I had const I still got some build issues where people tried to do arr[i][j] = stuff;, and there were const issues (I think the first operator[] returned a const which made the assignement impossible).

Reply:
Right now I'm changing things in my ale3d pull request to get the LEGACY flag passed to care properly, so please don't approve or merge this in until I have that resolved. There might be some other minor changes here, not sure

Reply:
Hi. I will keep this function. Please take another look, and tell me whether you want me to put the const in (in spite of issues when compatibility mode is turned on in ale3d)

Reply:
You can leave this as is.